cmd/ef_tests/state/runner_v2/result_check.rs

```diff
@@ -1,63 +1,102 @@
-pub fn create_report(
-    res: Result<(), RunnerError>,
-    test_case: &TestCase,
-    test: &Test,
-) -> Result<(), RunnerError> {
-    let report_path = PathBuf::from("./runner_v2/runner_report.txt");
-    let mut report = OpenOptions::new()
-        .append(true)
-        .create(true)
-        .open(report_path)
-        .unwrap();
-    let content = if res.is_ok() {
-        format!(
-            "Test checks succeded for test: {:?}, with fork {:?},  in path: {:?}.\n",
-            test.name, test_case.fork, test.path
-        )
-    } else {
-        format!(
-            "Test checks failed for test: {:?}, with fork: {:?},  in path: {:?}.\n",
-            test.name, test_case.fork, test.path,
-        )
-    };
-    let _ = report.write_all(content.as_bytes());
-
-    Ok(())
+pub struct PostCheckResult {
+    pub fork: Fork,
+    pub vector: (usize, usize, usize),
+    pub passed: bool,
+    pub root_dif: Option<(H256, H256)>,
+    pub accounts_diff: Option<Vec<AccountMismatch>>,
+    pub logs_diff: Option<(H256, H256)>,
+    pub exception_diff: Option<(Vec<TransactionExpectedException>, Option<VMError>)>,
+}
+impl Default for PostCheckResult {
+    fn default() -> Self {
+        Self {
+            fork: Fork::Prague,
+            vector: (0, 0, 0),
+            passed: true,
+            root_dif: None,
+            accounts_diff: None,
+            logs_diff: None,
+            exception_diff: None,
+        }
+    }
+}
+#[derive(Debug, Clone)]
+pub struct AccountMismatch {
+    pub address: Address,
+    pub expected_balance: U256,
+    pub actual_balance: U256,
+    pub expected_nonce: u64,
+    pub actual_nonce: u64,
+    pub expected_code: Bytes,
+    pub actual_code: Bytes,
+    pub expected_storage: HashMap<U256, U256>,
+    pub actual_storage: BTreeMap<H256, U256>,
 }
 
-/// Verify if the test has given the expected results: if an exception was expected, check it was the corresponding
+/// Verify if the test has reached the expected results: if an exception was expected, check it was the corresponding
 /// exception. If no exception was expected verify the result root.
 pub async fn check_test_case_results(
     vm: &mut VM<'_>,
     initial_block_hash: H256,
     store: Store,
     test_case: &TestCase,
     execution_result: Result<ExecutionReport, VMError>,
-) -> Result<(), RunnerError> {
-    if let Some(expected_exceptions) = test_case.post.expected_exception.clone() {
-        check_exception(expected_exceptions, execution_result)?;
+    genesis: Genesis,
+) -> Result<PostCheckResult, RunnerError> {
+    let mut checks_result = PostCheckResult {
+        fork: test_case.fork,
+        vector: test_case.vector,
+        ..Default::default()
+    };
+
+    if test_case.expects_exception() {
+        // Verify in case an exception was expected.
+        check_exception(
+            test_case.post.expected_exceptions.clone().unwrap(),
```

Comment:
Maybe we can use `if let Some(expected_exceptions) = test.post.expected_exceptions` and remove the need for the `unwrap`